weaviate/auth.py

```diff
@@ -1,102 +1,38 @@
+from typing import Optional
 
 
-class AuthCredentials(ABC):
-    """
-    Base class for getting the grant type and credentials.
-    """
-
-    def __init__(self):
-        self._credentials_body = {}
-
-    @abstractmethod
-    def get_credentials(self) -> dict:
-        """
-        Get credentials.
-        """
+@dataclass
+class AuthCredentials(metaclass=ABCMeta):
+    """Base class for authentication."""
 
 
+@dataclass
 class AuthClientCredentials(AuthCredentials):
-    """
-    Using a client secret for authentication.
-    In case of grant type client credentials.
-    """
-
-    def __init__(self, client_secret: str):
-        """
-        Using a client secret for authentication.
-        In case of grant type client credentials.
-
-        Parameters
-        ----------
-        client_secret : str
-            The client secret credentials, NOT access token.
-        """
+    """Authenticate for the client credential flow using client secrets.
 
-        super().__init__()
-        self._credentials_body["grant_type"] = "client_credentials"
-        self._client_secret_encoded = base64.b64encode(client_secret.encode("utf-8")).decode(
-            "utf-8"
-        )
-
-    def get_credentials(self) -> dict:
-        """
-        Get decoded credentials.
-
-        Returns
-        -------
-        dict
-            Decoded credentials.
-        """
+    Acquire the client secret from your identify provider and set the appropriate scope. Rhe client includes hardcoded
+    scopes for Azure, otherwise it needs to be supplied.
+    """
 
-        return_body = copy.deepcopy(self._credentials_body)
-        return_body["client_secret"] = base64.b64decode(
-            self._client_secret_encoded.encode("utf-8")
-        ).decode("utf-8")
-        return return_body
+    client_secret: str
+    scope: Optional[str] = None
 
 
+@dataclass
 class AuthClientPassword(AuthCredentials):
-    """
-    Using username and password for authentication.
-    In case of grant type password.
-    """
-
-    def __init__(self, username: str, password: str) -> None:
-        """
-        Using username and password for authentication.
-        In case of grant type password.
-
-        Parameters
-        ----------
-        username : str
-            The username to login with.
-        password : str
-            Password fot the given User.
-        """
+    """Using username and password for authentication. In case of grant type password."""
 
-        super().__init__()
-        self._credentials_body["grant_type"] = "password"
-        self._credentials_body["username"] = username
-        self._password_encoded = base64.b64encode(password.encode("utf-8")).decode("utf-8")
+    username: str
+    password: str
 
-    def get_credentials(self) -> dict:
-        """
-        Get decoded credentials.
 
-        Returns
-        -------
-        dict
-            Decoded credentials.
-        """
+@dataclass
+class AuthBearerConfig(AuthCredentials):
```

Comment:
This seems to be the same as `additional_headers`'s `Authorization` token. Do we want to have both of them?

Reply:
yep, I thought it is more userfrendly to put it at the same place as the other authorization methods, but didn't want to break any existing setups. Do you have an opinion here? :)

Reply:
I think we should have both options for now.

Reply:
oki, then let's leave it :)